scripts/sysdig-deploy/update-sysdig-deploy.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -e
+
+node_analyzer_chart_path="node-analyzer"
+agent_chart_path="agent"
+kspm_collectore_chart_path="kspm-collector"
+rapid_response_chart_path="rapid-response"
+admission_controller_chart_path="admission-controller"
+sysdig_deploy_path="sysdig-deploy"
+
+minor=0
+major=0
+patch=0
+
+check_update_needed () {
+    
+    new_subchart_version=$(yq eval '.version' charts/$chart/Chart.yaml)
+    sysdig_subchart_version=$(yq eval '.dependencies[] | select(.name == "'$chart'") | .version ' charts/$sysdig_deploy_path/Chart.yaml | sed "s/~//")
+
+    echo "subchart version in" $chart"/Chart.yaml"
+    echo $new_subchart_version
+    echo $chart "subchart version in sysdig-deploy/Chart.yaml"
+    echo $sysdig_subchart_version
+
+
+    IFS="."
+    read -ra prev_ver <<< "$sysdig_subchart_version"
+    read -ra next_ver <<< "$new_subchart_version"
+    
+    # update the subchart version in sysdig-deploy/Chart.yaml
+    yq -i '( .dependencies[] | select(.name == "~'$chart'") | .version) = "'"$new_subchart_version"'"' charts/$sysdig_deploy_path/Chart.yaml
+
+    for ((idx=0; idx<3; ++idx)); do
+        if [ ${next_ver[idx]} -gt ${prev_ver[idx]} ] 
+        then
+            if [ $idx -eq 0 ]
+            then
+                major=1
+            elif [ $idx -eq 1 ]
+            then
+                minor=1
+            elif [ $idx -eq 2 ]
+            then
+                patch=1
+            fi
+        fi
+    done
```

Comment:
Feel free to ignore the following
```suggestion
    [ ${next_ver[0]} -gt ${prev_ver[0]} ] && major = 1
    [ ${next_ver[1]} -gt ${prev_ver[1]} ] && minor = 1
    [ ${next_ver[2]} -gt ${prev_ver[2]} ] && patch = 1
```

Reply:
I tried this change and got this error so ill probably skip this code change
```
yash@yash charts % ./scripts/sysdig-deploy/update-sysdig-deploy.sh 
subchart version in node-analyzer/Chart.yaml
1.7.33
node-analyzer subchart version in sysdig-deploy/Chart.yaml
1.7.32
patch: **** Can't open patch file 1 : No such file or directory
```

Reply:
Oh I see, typical mistake.
There needs to be no space around the equal sign, i.e. patch=1

Reply:
I tried this change again after your correction
```
--- a/scripts/sysdig-deploy/update-sysdig-deploy.sh
+++ b/scripts/sysdig-deploy/update-sysdig-deploy.sh
@@ -29,21 +29,10 @@ check_update_needed () {
     # update the subchart version in sysdig-deploy/Chart.yaml
     yq -i '( .dependencies[] | select(.name == "'$chart'") | .version) = "~'"$new_subchart_version"'"' charts/$sysdig_deploy_path/Chart.yaml
 
-    for ((idx=0; idx<3; ++idx)); do
-        if [ ${next_ver[idx]} -gt ${prev_ver[idx]} ] 
-        then
-            if [ $idx -eq 0 ]
-            then
-                major=1
-            elif [ $idx -eq 1 ]
-            then
-                minor=1
-            elif [ $idx -eq 2 ]
-            then
-                patch=1
-            fi
-        fi
-    done
+    [ ${next_ver[0]} -gt ${prev_ver[0]} ] && major = 1
+    [ ${next_ver[1]} -gt ${prev_ver[1]} ] && minor = 1
+    [ ${next_ver[2]} -gt ${prev_ver[2]} ] && patch = 1
+
 }
 
 charts=( "$node_analyzer_chart_path" "$agent_chart_path" "$kspm_collector_chart_path" "$rapid_response_chart_path" "$admission_controller_chart_path" )
yash@yash charts % ./scripts/sysdig-deploy/update-sysdig-deploy.sh
subchart version in node-analyzer/Chart.yaml
1.7.33
node-analyzer subchart version in sysdig-deploy/Chart.yaml
1.7.33
```
the execution stops after the first chart